### PR TITLE
Jupyter notebook support

### PR DESCRIPTION
**Pull request by Chadsr**
Since https://github.com/psf/black/pull/2357 was merged, `black` supports linting Jupyter notebook files (`.ipynb`).
This PR therefore aims to extend the action to also annotate Jupyter notebook files.

I'm not sure how this will play with Github's displaying of Jupyter notebook code, or whether the output format from black for Jupyter is different from normal python file output, so this may need more testing/adaptation.

P.S. I debated adding an extra `with` arg for this, but then figured it's easier to just use the existing `black_args`  , if you wish to exclude/disable `.ipynb` checks, but maybe a README remark on this is useful?

Input/opinions/edits welcome.

### CODE REVIEW

**Comment by Chadsr** [comment]
I couldn't manage to test this properly on my local machine due to #45, so please feel free to ping me if there test/build issues that still need resolving.

**Comment by rickstaa** [comment]
@Chadsr Thanks a lot for implementing this pull request! I noticed you transferred your pull request to a draft. Feel free to let me know when you are ready. I added one small commit to account for the false positives received from the spellcheck linter.

**Comment by Chadsr** [comment]
@rickstaa I've added some Bash magic, so that the test actually checks the output of black is as expected.

Beyond that, i'm not yet totally sure how to verify if reviewdog handles the `.ipynb` annotations "properly" in Github?

**Comment by Chadsr** [comment]
Either I am missing something, or the `.ipynb` file annotation does not work properly, yet (probably due to structural differences in the file vs black output).

I would expect to see the reviewdog annotations added to this pull request, correct?

**Comment by Chadsr** [comment]
Should we maybe come to a decision, as to whether to put this PR on hold until annotations are definitely working properly, or do merge this and open an issue/PR to figure out the absolute vs relative line number problem?

Having some form of output as seen in https://github.com/reviewdog/action-black/actions/runs/1562786562 still seems like a good step in the right direction, and possibly worth of merging as-is, for now.

**Comment by rickstaa** [comment]
@Chadsr Thanks again for your pull request. I haven't yet been able to find the time to review your pull request properly. I try to do it this or next weekend. You can also try requesting a review from the other maintainers to speed the process along.

**Comment by Chadsr** [comment]
@rickstaa Whoops, this PR slipped my mind. No problem.

Technically I can merge this myself, now that I joined the reviewdog org, but it would still be good to get a second opinion, I think?

**Comment by rickstaa** [comment]
@Chadsr I understand! I will try to squeeze in some time to review it this weekend.

**Comment by Chadsr** [comment]
On the subject of dedicating time to this repo, I should also add, that moving forward I will happily take on a pro-active maintainer role to spread the load.

**Comment by rickstaa** [comment]
> On the subject of dedicating time to this repo, I should also add, that moving forward I will happily take on a pro-active maintainer role to spread the load.
> 
> * and on this subject, maybe it's useful for us to add a `.github/CODEOWNERS` file so that contributors can more easily know who to ping, or even better, just let Github automatically assign reviewers to PRs.

That's very nice! 🚀 I also like your ideas. I, unfortunately, still haven't found the time to review the pull request but I will look at it this weekend.

**Comment by rickstaa** [comment]
> @rickstaa I've added some Bash magic, so that the test actually checks the output of black is as expected.
> 
> Beyond that, i'm not yet totally sure how to verify if reviewdog handles the `.ipynb` annotations "properly" in Github?
> 
> Edit: whoops, now I realise that the new test only covers pushes, not PRs. Let me fix that up. Edit 2: Actually, since `test-check` always runs, even during the `pull_request` event, I think this might be sufficient as is?

@Chadsr You are right the new test should be enough I like it! I extended it a bit to test both files that should be checked and the ones that should not be checked (see 
3dc3b32).

**Comment by rickstaa** [comment]
@Chadsr I reviewed your pull request this morning, the code looks very clean and it works (see https://github.com/rickstaa/action-test-repo/actions/runs/1822872981). I agree that we might be able to improve the review dog output later but I think for now your pull request already is very valuable. Thanks again for implementing it! 🚀

**Review by rickstaa — APPROVED** [review]

**Comment by rickstaa** [comment]
@Chadsr Feel free to merge it after you reviewed 89696f9.

**Comment by rickstaa** [comment]
@Chadsr Thanks for merging this pull request.

Please be aware that with the current release action https://github.com/reviewdog/action-black/actions/workflows/release.yml the GitHub marketplace version does not automatically update. As a result, after each release, we have to manually click the edit button and save it for the version to update in the marketplace.

I still have to change this to using the `gh` https://github.com/cli/cli (see https://github.com/reviewdog/reviewdog/issues/1043), but I haven't found the time yet. I will created a pull request to remind us to change this.

**Pull request merged**
